ansible/roles/images/tasks/check-or-pull-images.yaml

```
@@ -0,0 +1,16 @@
+---
+# If image is present, its name is written to stdout. Return code is 0.
+#   > ctr -n k8s.io images check -q name==docker.io/mesosphere/pause-alpine:3.2; echo $?
+#   docker.io/mesosphere/pause-alpine:3.2
+#   0
+#
+# If image is not present, or is present, but some layer is missing, its name is not written to stdout. Return code is 0.
+#   > ctr -n k8s.io images check -q name==docker.io/mesosphere/pause-alpine:999; echo $?
+#   0
+- name: check existing image to ensure all content is available locally
+  shell: "ctr -n k8s.io image check -q name=={{ image_name }}"
+  register: ctr_image_check
+
+- name: pull image
+  shell: "ctr -n k8s.io image pull {{ image_name }}"
+  when: ctr_image_check.stdout == "{{ image_name }}"
```

```suggestion
  when: ctr_image_check.stdout != image_name
```
OR if there are some newlines or spaces in stdout
```suggestion
  when: 'image_name not in ctr_image_check.stdout'
```

---

Fixed